src/db/data_source.h

```diff
@@ -0,0 +1,43 @@
+#ifndef SRC_DB_DATA_SOURCE_H_
+
+#define SRC_DB_DATA_SOURCE_H_
+
+#include <string>
+
+#include "absl/status/status.h"
+
+namespace db {
+
+// Abstract Database Interface that supports transaction operations.
+// TODO(n-triple-a): Implement interface using LMDB primitives
+class DataSource {
+ public:
+  DataSource() = default;
+  ~DataSource() = default;
+  
+  // Connects to the database instance.
+  virtual absl::Status Connect();
+
+  // Begins a transaction.
+  virtual absl::Status Begin();
```

Comment:
Should this returns a transaction id which used in later `Commit` and `Abort` so we can have concurrent transactions?

Reply:
Same for `Get` and `Put` as well.

Reply:
I was thinking something like below. Let me know what you think:

```
int tid = db.Begin();
db.Put(tid, “key1”, 123);
db.Get(tid, “key1”);
db.Commit(tid);
```

Reply:
There are two options to implement this. 
1) Here, the intended use case is for each DataSource, it only maintains a single transaction. So to support parallel txn, it's up-to callsite to maintain the <txn_id, DataSource> map.

2) For your proposal, there is a single DataSource object that manages multiple txns in parallel. I probably want to avoid this option to prevent possible race condition on the DataSource's internal state.

But let me clearly document the intended usage in the followup PR and discuss it there. Thanks for asking.

Reply:
Who is responsible for determining which transactions have which locks? Is it the DataSource or the callsite? Either is fine, but we should decide on it.

The transaction id is assigned by the coordinator. Could we use that one instead of creating a new one?

Reply:
Right, I think we probably want to align on the questions raised here - the definition of a transaction and data source, the lock granularity, the responsibilities of a DataSource and its callsite.